test/test_server.py

```diff
@@ -175,6 +177,37 @@ def test_daemon_double_start(self):
         cfg_file = self.resource('src/ngamsCfg.xml')
         self.assertNotEqual(0, self._run_daemon_start(self.point_to_ngas_root(cfg_file)))
 
+    def test_daemon_stale_pid_file(self):
+        # Create a stale PID file, it should refuse to start
+        # Define a root directory for creating the temporary NGAS volume
+        root_path = tempfile.mkdtemp(prefix="ngas_")
+        run_path = os.path.join(root_path, "var", "run")
+        os.makedirs(run_path)
+        # Create a stale PID lock file with a bogus PID
+        pid_path = os.path.join(run_path, "ngamsDaemon.pid")
+        with open(pid_path, "w") as pid_file:
+            pid_file.write("99999999")
+        # self.prepExtSrv(daemon=True, delDirs=0, root_dir=root_path)
+        self.assertRaises(BaseException, self.prepExtSrv, daemon=True, delDirs=0, root_dir=root_path)
+        # Clean up the temporary volume directories and files
+        shutil.rmtree(root_path, True)
+
+    def test_daemon_stale_pid_file_with_force(self):
+        # Create a stale PID file but set '-force' command line option,
+        # it should clean up the old PID file and forcibly restart if necessary
+        root_path = tempfile.mkdtemp(prefix="ngas_")
+        run_path = os.path.join(root_path, "var", "run")
+        os.makedirs(run_path)
+        # Create a stale PID lock file with a bogus PID
+        pid_path = os.path.join(run_path, "ngamsDaemon.pid")
+        with open(pid_path, "w") as pid_file:
+            pid_file.write("99999999")
+        self.prepExtSrv(daemon=True, delDirs=0, force=True, root_dir=root_path)
+        self.assertEqual(0, self._run_daemon_status(self.extSrvInfo[-1].cfg_file))
+        self.terminateAllServer()
+        # Clean up the temporary volume directories and files
+        shutil.rmtree(root_path, True)
+
```

Comment:
The two tests here share a bit of code to prepare for the actual tests, could that be shared between the two?

Reply:
My apologies. You are correct. I should have committed the pep-8 changes and the issue-54 changes in two separate commits. I have now refactored the unit test code so the shared code is now shared in a common method.